### PR TITLE
Enable arm64 build for Arrow

### PR DESCRIPTION
**Pull request by jgiannuzzi**
**Describe the pull request**

- #### What does your PR fix?  
  This PR adds support for building Arrow for arm64, and for cross-compilation to work (e.g. building `arm64-osx` from a `x64-osx` host). Building on `arm64-windows` is still not supported and would require a patch to Arrow (maybe for a future PR?).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Adding non-Windows `arm64` triplets to the current list (all `x64` triplets).
  No change to the CI baseline (The non-Windows `arm64` triplets are community supported anyway).

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by jgiannuzzi** [comment]
How can I tell the CI to build the arrow port on `arm64-windows`?

File: ports/arrow/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "arrow",
   "version": "5.0.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Cross-language development platform for in-memory analytics",
   "homepage": "https://arrow.apache.org",
   "supports": "x64",
```
Comment (autoantwort) [inline ports/arrow/vcpkg.json:7]: You have to change this line to `x64 | arm64`

Comment (jgiannuzzi) [reply ports/arrow/vcpkg.json]: Thank you @autoantwort!

File: ports/arrow/portfile.cmake [hunk 2]
```diff
@@ -1,4 +1,4 @@
-vcpkg_fail_port_install(ON_ARCH "x86" "arm" "arm64")
+vcpkg_fail_port_install(ON_ARCH "x86" "arm")
```
Comment (JackBoosY) [inline ports/arrow/portfile.cmake:1]:
```suggestion
```
Since keyword `supports` have implement this, we don't need this anymore.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/arrow/portfile.cmake (hunk 2 above)
Comment (jgiannuzzi) [reply ports/arrow/portfile.cmake]: Done. Thanks @JackBoosY!

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM.

**Pull request merged**